qml/pages/GalleryUI.qml

```diff
@@ -1,22 +1,100 @@
 import QtQuick 2.0
 import Sailfish.Silica 1.0
+import Nemo.FileManager 1.0
 
 Page {
     id: galleryPage
 
-    property var photoList: ({})
+    property var fileList: ({})
 
     // The effective value will be restricted by ApplicationWindow.allowedOrientations
     allowedOrientations: Orientation.All
 
+    backNavigation: false
+
+    function removeFile(idx) {
+        var path = fileList.get(idx).filePath
+        console.log("Removing", path)
+        FileEngine.deleteFiles( [ path ] )
+        fileList.remove(idx)
+        if (gallery.count === 0) {
+            console.log("Closing empty gallery!")
+            pageStack.pop()
+        }
+    }
+
+    Rectangle {
```

Comment:
Probably simpler to have button directly in Row or IconButtons directly and use anchors to do the layout instead of using extra Rectangles for layout.